src/fs_event_wrap.cc

```diff
@@ -156,7 +160,36 @@ void FSEventWrap::OnEvent(uv_fs_event_t* handle, const char* filename,
   };
 
   if (filename != nullptr) {
-    argv[2] = OneByteString(env->isolate(), filename);
+    size_t len = strlen(filename);
+    // If encoding is BUFFER, return a Buffer.
+    // If encoding is UCS2, attempt to decode as UCS2 or report an error.
+    // Otherwise, Encode as the requested type, with UTF8 as the fallback.
+    if (wrap->encoding_ == BUFFER) {
+      argv[2] = Buffer::Copy(env, filename, len).ToLocalChecked();
+    } else if (wrap->encoding_ == UCS2) {
+      // Note: this is lossy. If the filename is not aligned at 2 bytes,
+      // the last byte in the filename will be silently dropped in the 
+      // copy.
```

Comment:
Where we going to either error here or default back to UTF8? I feel like if the filename isn't dividable by 2 then we can assume the wrong file encoding was passed.